pkg/controller/replication.go

```diff
@@ -126,10 +126,62 @@ func (s *Service) CreateStorageProtectionGroup(ctx context.Context,
 	}
 
 	if protocol == "nfs" { // NFS protocol indicates a FileSystem on the storage array and requires different handling from block Volumes.
-		return nil, status.Error(codes.InvalidArgument, "replication is not supported for NFS volumes")
+		// get the NAS server from the filesystem
+		fs, err := arr.GetClient().GetFS(ctx, id)
+		if err != nil {
+			return nil, err
+		}
+		nasId := fs.NasServerID
+		log.Infof("!!!!!!!!!")
+		log.Infof("NAS ID: " + nasId)
```

Comment:
With Infof(), format can be specified. Ex: ("NAS ID: %s", nasId). Please update at other places and remove above test log line.

Reply:
Most of these log lines were left in by mistake. I'll go and clean out the ones that aren't beneficial to end-user and were placed for testing/development, and fix any uses of Infof() that remain.